File: Quizzical/src/main/java/com/asfour/models/ApiResponse.java
package com.asfour.models;

import com.google.gson.annotations.SerializedName;

/**
 * Created by waqqas on 6/17/16.
 */
public class ApiResponse<T> {
    @SerializedName("Data")
    private T data;
    @SerializedName("CurrentPage")
    private int currentPage;
    @SerializedName("TotalPages")
    private int totalPages;
    @SerializedName("PageSize")
    private int pageSize;

    public T getData() {
        return data;
    }

    public int getCurrentPage() {
        return currentPage;
    }

    public int getTotalPages() {
        return totalPages;
    }

    public int getPageSize() {
        return pageSize;
    }
}

File: Quizzical/src/main/java/com/asfour/data/api/ApiResponse.kt
package com.asfour.data.api

import com.google.gson.annotations.SerializedName

data class ApiResponse<out T>(@SerializedName("data") val data: T,
                              @SerializedName("page") val page: Int,
                              @SerializedName("size") val size: Int,
                              @SerializedName("page_count") val pageCount: Int,
                              @SerializedName("last") val last: Boolean)